src/db/controller/channel.ts

```diff
@@ -0,0 +1,328 @@
+      connection_id: connection.counterparty_connection_id,
+      port_id: event.channelOpenCloseInfo.srcPortId,
+      channel_id: event.channelOpenCloseInfo.srcChannelId,
+      counterparty_chain_id: chainId,
+      counterparty_connection_id: event.channelOpenCloseInfo.dstConnectionId,
+      counterparty_port_id: event.channelOpenCloseInfo.dstPortId,
+      counterparty_channel_id: event.channelOpenCloseInfo.dstChannelId,
+    }
+
+    return () => {
+      del<ChannelOpenCloseTable>(DB, this.tableName, [
+        {
+          state: ChannelState.INIT,
+          counterparty_chain_id: connection.counterparty_chain_id,
+          counterparty_port_id: channelOnOpen.port_id,
+          counterparty_channel_id: channelOnOpen.channel_id,
+        },
+      ]) // remove init
+      insert(DB, this.tableName, channelOnOpen)
+    }
+  }
+
+  private static async feedChannelOpenAckEvent(
+    rest: RESTClient,
+    chainId: string,
+    event: ChannelOpenCloseEvent
+  ): Promise<() => void> {
+    const connection = await ConnectionController.getConnection(
+      rest,
+      chainId,
+      event.channelOpenCloseInfo.srcConnectionId
+    )
+
+    // add channel on open for dst chain
+    const channelOnOpen: ChannelOpenCloseTable = {
+      in_progress: Bool.FALSE,
+      height: event.channelOpenCloseInfo.height,
+      state: ChannelState.ACK,
+      chain_id: connection.counterparty_chain_id,
+      connection_id: connection.counterparty_connection_id,
+      port_id: event.channelOpenCloseInfo.dstPortId,
+      channel_id: event.channelOpenCloseInfo.dstChannelId,
+      counterparty_chain_id: chainId,
+      counterparty_connection_id: event.channelOpenCloseInfo.srcConnectionId,
+      counterparty_port_id: event.channelOpenCloseInfo.srcPortId,
+      counterparty_channel_id: event.channelOpenCloseInfo.srcChannelId,
+    }
+
+    return () => {
+      del<ChannelOpenCloseTable>(DB, this.tableName, [
+        {
+          state: ChannelState.TRYOPEN,
+          counterparty_chain_id: channelOnOpen.chain_id,
+          counterparty_port_id: channelOnOpen.port_id,
+          counterparty_channel_id: channelOnOpen.channel_id,
+        },
+      ]) // remove open try
+      del<ChannelOpenCloseTable>(DB, this.tableName, [
+        {
+          state: ChannelState.INIT,
+          counterparty_chain_id: chainId,
+          counterparty_port_id: channelOnOpen.counterparty_port_id,
+          counterparty_channel_id: channelOnOpen.counterparty_channel_id,
+        },
+      ]) // remove init
+      insert(DB, this.tableName, channelOnOpen)
+    }
+  }
+
+  private static async feedChannelOpenConfirmEvent(
+    rest: RESTClient,
+    chainId: string,
+    event: ChannelOpenCloseEvent
+  ): Promise<() => void> {
+    const connection = await ConnectionController.getConnection(
+      rest,
+      chainId,
+      event.channelOpenCloseInfo.dstConnectionId
+    )
+    return () => {
+      del<ChannelOpenCloseTable>(DB, this.tableName, [
```

Comment:
hmm is this for cleanup?? 

in the flow, I thought each channel delete an event created by counterparty chain.